packages/nimble-components/src/text-area/tests/text-area.spec.ts

```diff
@@ -79,35 +79,25 @@ describe('Text Area', () => {
         { name: 'aria-roledescription' }
     ];
     describe('should reflect value to the internal control', () => {
-        const focused: string[] = [];
-        const disabled: string[] = [];
-        for (const attribute of attributeNames) {
-            const specType = getSpecTypeByNamedList(
-                attribute,
-                focused,
-                disabled
-            );
-            // eslint-disable-next-line @typescript-eslint/no-loop-func
-            specType(`for attribute ${attribute.name}`, async () => {
+        parameterizeNamedList(attributeNames, (spec, name, value) => {
```

Comment:
I just picked on at random, but can you do a pass on the changes to make sure the lists that are passed in are `as const`

In this case:
```
    const attributeNames: {
        name: string,
        value?: string,
        boolean?: boolean
    }[] = [...];
```

should remove the general type and end up as:

```
    const attributeNames = [...] as const;
```

that's the intention of the second bullet in the tech debt description duplicated as follows:
> Migrate tests in such a way as they take advantage of the strict typing (using as const on the test cases) and consistent names for the test function parameters (spec, name, and value)

Reply:
There were a few places where I couldn't make the input `const` because the input was calculated based on some other data structure (e.g. icons and tokens), but I updated all the rest.

This required a few changes to non-test code where function parameters needed to be marked as `readonly`, so I made those updates as well.

Reply:
looks good!